### ISSUE TITLE
Boosting Deep Learning Risk Prediction with Generative Adversarial Networks for Electronic Health Records

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#111]
**일명 ehrGAN 논문.
지난번에 읽었던 논문의 related work로 기재되어 있었고 설명이 흥미로워 찾아 읽음.**

## Abstract
* boost risk prediction performance with limited EHR data
* provide plausible labeled EHR data by mimicking real patient records
* -> augment the training dataset in a semi-supervised learning manner
* onset prediction performance를 향상시키기 위해 GAN과 CNN을 함께 사용함.

## Introduction
* "most importantly saving lives" (논문 읽다가 감동적이어서..)
* EHR data 문제점
  * the amount of clinical data, especially with accurate labels and for rare diseases and conditions, is somewhat limited and far from most models’ requirements
  * semi-supervised learning (SSL)로 overcome.
* propose a generative model, ehrGAN, for EHR data via adversarial training, which is able to generate convincing and useful samples similar to realistic patient records in existing data
* propose a semi-supervised learning framework which achieves boosted risk prediction performance by utilizing the augmented data and representations from the proposed generative models

## Related Work
* 머신러닝과 일반 GAN 설명..

## The Proposed Model
* ehrGAN 모델 그림.
  <image src="https://user-images.githubusercontent.com/67627471/131139341-606a575a-1527-4d72-bb02-c0cb69d830c3.png" width="60%" height="60%" />

* 베이직 모델로 CNN 사용.
  * 1D convolutional layer & temporal dimension(embedding 의미하는 듯) & max over-time pooling layer
  * 인풋은 EHR records of patient p (temporal embedding matrix)
  * The embedding for medical events is trained by Word2vec model
  * multiple levels에서 temporal dependencies를 잡아내기 위해 combination of filters with different lengths 사용함.
  * apply a max-pooling operation along the temporal dimension to keep the most important features across the time
  * temporal pooling converts the inputs with different temporal lengths into a fixed length output vector
  * 마지막으로 fully connected soft-max layer => prediction probabilities
* ehrGAN
  * G and D falls into the family of 1D convolutional neural networks (CNNs) and 1D deconvolutional neural networks (DCNNs)
  * Discriminator
    *  adopt the structure of the basic prediction model to the discriminator
    * replace the top prediction layer by a single sigmoid unit to output the probability of the input data being drawn from the real dataset
  * Generator
    * encoded by a de-convolutional neural network with two consecutive fully connected layers
    * reshaped and followed by two de-convolutional layers to perform upsampling convolution
    * (문제점) can not be directly used in semi-supervised learning setting as the model is trained only to differentiate real or
synthetic data instead of the classes
    * (해결책) Generator with variational contrastive divergence
    * VCD 기반.
    * distribution을 직접적으로 학습하지 않고, transition distribution을 학습함.
    * <image src="https://user-images.githubusercontent.com/67627471/131142562-b7f9e38f-ad80-42e5-b90b-e3a3b6d37b53.png" width="50%" height="50%" />

* ehrGAN 최적화 알고리즘
  <image src="https://user-images.githubusercontent.com/67627471/131142144-cd7fe1ab-0ac8-456d-9ecd-ee239f3d545c.png" width="50%" height="50%" />

* 학습된 ehrGAN으로 SSL 진행.
* SSL GAN 손실 함수 설정.
  <image src="https://user-images.githubusercontent.com/67627471/131142921-649847d3-bce5-4cc4-8f4c-0d75f8331970.png" width="40%" height="40%" />

## Datasets
* real-world longitudinal Electronic Health Record database from health insurance company
* 218,680 patients
*14,969,489 observations of 14, 690 unique medical events
* 2011-2015년
* Congestive heart failure (Heart Failure); 3, 357 confirmed patients in case group and 6, 714 patients in control group
* Diabetes (Diabetes); 2, 248 patients in case group and 4, 496 patients in control group

## Results
* prediction 성능 비교.
  <image src="https://user-images.githubusercontent.com/67627471/131142302-729a823e-6572-4e85-b013-a84eba0a1650.png" width="40%" height="40%" />

* PERFORMANCE COMPASSION OF DIFFERENT CNN AND SSL PREDICTION MODELS ON FOUR SUB-DATASETS.
  <image src="https://user-images.githubusercontent.com/67627471/131142412-a2ceefdb-5a88-4dc5-93fb-3b5ed590dd10.png" width="70%" height="70%" />

* AUROC SCORE COMPARISON WITH DIFFERENT ρ
  <image src="https://user-images.githubusercontent.com/67627471/131142483-4b22a593-7854-4e96-8c43-c2b9d1ef3068.png" width="40%" height="40%" />

##
* 이제까지 읽은 ehr 관련 GAN 논문 다 모아서 정리 한 번 해야겠음.
